### ISSUE TITLE
Error types should not implement `source` _and_ print inner error in Display

### ISSUE THREAD

**Issue by jonhoo** [awslabs/smithy-rs#1345]
`SdkError`, and possibly other error types within the SDK, currently expose inner errors both through `Error::source` _and_ `fmt::Display`:

https://github.com/awslabs/smithy-rs/blob/692bf94ba30f570365f559c67d6a9981c037c371/rust-runtime/aws-smithy-http/src/result.rs#L176-L206

The recommendation in the ecosystem is to _either_ return inner errors through `Error::source` _or_ print the inner error in `Display`, but never both (see https://github.com/rust-lang/project-error-handling/issues/27#issuecomment-763950178, https://github.com/rust-lang/project-error-handling/issues/23, https://github.com/rust-lang/api-guidelines/pull/210), as otherwise errors end up being duplicated whenever error reporters print error backtraces. For example, I recently got an error trace that ended in:

```
    6: failed to construct request: Failed to load credentials from the credentials provider: An error occurred while loading credentials: io error: error trying to connect: invalid certificate: UnknownIssuer
    7: Failed to load credentials from the credentials provider: An error occurred while loading credentials: io error: error trying to connect: invalid certificate: UnknownIssuer
```

**Comment by jonhoo** [comment]
The same is the case for `CredentialsError`:

https://github.com/awslabs/smithy-rs/blob/a46e4899cc854e02e2c25d20797a497c67ad28f4/aws/rust-runtime/aws-types/src/credentials/provider.rs#L106-L144

**Comment by rcoh** [comment]
yeah this is definitely the case currently—is there a plan to add backtrace reporting by default? I worry about customers not using a "fancy" printer missing error context

**Comment by rcoh** [comment]
I guess this may guide us towards "use display but not source"?

**Comment by jdisanti** [comment]
Implemented as part of #1926
